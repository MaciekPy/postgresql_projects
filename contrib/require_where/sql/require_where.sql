--
--   Test require_where
--

\set echo all

CREATE TABLE test_require_where(t TEXT);

UPDATE test_require_where SET t=t; -- succeeds

DELETE FROM test_require_where; -- succeeds

LOAD 'require_where';

UPDATE test_require_where SET t=t; -- fails

UPDATE test_require_where SET t=t WHERE true; -- succeeds

DELETE FROM test_require_where; -- fails

DELETE FROM test_require_where WHERE true; -- succeeds
