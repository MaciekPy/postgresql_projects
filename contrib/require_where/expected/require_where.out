--
--   Test require_where
--
\set echo all
CREATE TABLE test_require_where(t TEXT);
UPDATE test_require_where SET t=t; -- succeeds
DELETE FROM test_require_where; -- succeeds
LOAD 'require_where';
UPDATE test_require_where SET t=t; -- fails
ERROR:  UPDATE requires a WHERE clause when the "require_where" extension is loaded.
HINT:  To update all rows, use "WHERE true" or similar.
UPDATE test_require_where SET t=t WHERE true; -- succeeds
DELETE FROM test_require_where; -- fails
ERROR:  DELETE requires a WHERE clause when the "require_where" extension is loaded.
HINT:  To delete all rows, use "WHERE true" or similar.
DELETE FROM test_require_where WHERE true; -- succeeds
